### DISCUSSION TITLE
@types/node: Cannot export 'Buffer'

### DISCUSSION THREAD

**Discussion by iliubinskii** [DefinitelyTyped/DefinitelyTyped#54260] [Issues with a @types package]
Getting typescript error with v16.0.0:
node_modules/@types/node/buffer.d.ts: Cannot export 'Buffer'. Only local declarations can be exported from a module.

Error disappears after downgrading to v15.12.5.

**Comment by KapitanOczywisty** [comment]
This exact error is failing CI for my PR #54270 Cause can be found here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53669/files#diff-2ffb40bea23d1727aab4bc7b8bebc17d57fc1e3521fdf43de64855dbf56483e6

@rbuckton Since you've been involved in the review process can you take a look at that?

**Comment by SimonSchick** [comment]
Needs more info, what version of TS are you using?

**Reply by KapitanOczywisty** [reply]
You can easily reproduce this by running `npm test pouchdb-core` on up-to-date master.  Typescript 4.4

https://github.com/DefinitelyTyped/DefinitelyTyped/commit/0248e9dec042972d3be9d2ded27b1576934cd8a7/checks?check_suite_id=3153630796

**Reply by iliubinskii** [reply]
I am using latest typescript 4.4. The project that raises error uses pouchdb. So, I guess KapitanOczywisty's reproduction covers my issue.

**Comment by SimonSchick** [comment]
Strange, other packages that depend on node do not have these issues, I will look into it a bit.

**Comment by SimonSchick** [comment]
`pouchdb-core` is re-declaring `Buffer` and `Blob` globals, this needs to be removed, this is not an issue with node typings.

**Reply by SimonSchick** [reply]
Sorry new to discussions, I will format this better in the future.

**Reply by KapitanOczywisty** [reply]
Error message is broken then. 
And seems like they did that on purpose https://github.com/DefinitelyTyped/DefinitelyTyped/commit/74b5c85d67b98bcb682f76b2e66442c66c03eba0 I'm not entirely sure how to deal with this...

**Reply by KapitanOczywisty** [reply]
@SimonSchick I've made PR if you want to take a look #54278

**Reply by TheSlimvReal** [reply]
Applying the changes from #54478 does not solve the issue for me locally.

**Reply by KapitanOczywisty** [reply]
@TheSlimvReal #54479 should fix this issue

**Comment by TheSlimvReal** [comment]
The compiler error can be temporarily fixed by adding `"skipLibCheck": true` to the `tsconfig.json` file.

**Comment by bravier** [comment]
Hi,

I'm also impacted by this `Cannot export 'Buffer'` error while I'm using latest `@types/pouchdb-core` (7.0.8 - which includes PR #54478) and `@types/node` (16.4.3 - which includes PR #54479).

Is there any workaround other than hiding the compiler error or locking my `@types/node` version?

Thank you.

**Reply by KapitanOczywisty** [reply]
#54478 was only fixing failing tests. The other PR should fix this, but problem still exists.. @sandersn 
Would be probably good idea to have another set of tests with node.

**Reply by sandersn** [reply]
I came up with a possible solution at https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54812
